api/src/main/java/gov/cms/ab2d/api/controller/BulkDataAccessAPI.java

```diff
@@ -124,14 +125,29 @@
         checkResourceTypesAndOutputFormat(resourceTypes, outputFormat);
         checkSinceTime(since);
 
-        Job job = jobService.createJob(resourceTypes, ServletUriComponentsBuilder.fromCurrentRequest().toUriString(),
-                null, outputFormat, since);
+        Job job = jobService.createJob(resourceTypes, getCurrentUrl(), null, outputFormat, since);
 
         logSuccessfulJobCreation(job);
 
         return returnStatusForJobCreation(job);
     }
 
+    private String getCurrentUrl() {
+        String request = ServletUriComponentsBuilder.fromCurrentRequest().toUriString();
+        request = request.replace(":80/", "/");
+        if (!shouldReplaceWithHttps()) {
+            return request;
+        }
+        request = request.replace("http://", "https://");
```

Comment:
If you'd like to do this replacement in a more OOP way, `org.springframework.web.util.UriComponentsBuilder#scheme` method will do the trick

Reply:
Ah, thanks. That is what I was looking for. I just spent so much time googling for it unsuccessfully, that I figured I would put it out there. Maybe that will solve it coming back with :80/